### PR TITLE
Use Python Version: Add input architecture to error message when a version isn't found

### PR DESCRIPTION
**Pull request by brcrista**
Also made some small UI improvements:
image

**Testing**
* L0
* Inspect UI in designer
* Manually on hosted agent

### CODE REVIEW

File: Tasks/UsePythonVersionV0/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -5,13 +5,13 @@
   "loc.instanceNameFormat": "Use Python $(versionSpec)",
   "loc.group.displayName.advanced": "Advanced",
   "loc.input.label.versionSpec": "Version spec",
-  "loc.input.help.versionSpec": "Version range or exact version of a Python version to use.",
+  "loc.input.help.versionSpec": "Version range or exact version of a Python version to use, using semver's version range syntax. [More information](https://www.npmjs.com/package/semver)",
```
Comment (brcrista) [inline Tasks/UsePythonVersionV0/Strings/resources.resjson/en-US/resources.resjson]: I would have used an fwlink, but the new aka.ms tool is broken.  At any rate, when we write our own documentation I will probably rewrite this string to remove the reference to semver.

File: Tasks/UsePythonVersionV0/main.ts [hunk 2]
```diff
@@ -14,7 +14,6 @@ import { usePythonVersion } from './usepythonversion';
         getPlatform());
         task.setResult(task.TaskResult.Succeeded, "");
     } catch (error) {
-        task.error(error.message);
```
Comment (brcrista) [inline Tasks/UsePythonVersionV0/main.ts]: This was causing a duplicate error message to be logged.

File: Tasks/UsePythonVersionV0/task.json [hunk 3]
```diff
@@ -30,8 +30,8 @@
             "type": "string",
             "label": "Version spec",
             "required": true,
-            "defaultValue": ">= 3.3",
-            "helpMarkDown": "Version range or exact version of a Python version to use."
+            "defaultValue": "3.x",
```
Comment (brcrista) [inline Tasks/UsePythonVersionV0/task.json]: Synced with @DavidStaheli on this, we think this is cleaner.  The semantics are essentially the same as >= 3.3 (Pythons before 3.3 are very old and were essentially beta releases of Python 3.  3.4 is the oldest version we are currently shipping on an agent.)

Comment (bryanmacfarlane) [reply Tasks/UsePythonVersionV0/task.json]: +1

File: Tasks/UsePythonVersionV0/task.json [hunk 4]
```diff
@@ -12,7 +12,7 @@
     "author": "Microsoft Corporation",
     "version": {
         "Major": 0,
-        "Minor": 136,
+        "Minor": 137,
```
Comment (madhurig) [inline Tasks/UsePythonVersionV0/task.json]: We are in S138

Comment (brcrista) [reply Tasks/UsePythonVersionV0/task.json]: Whoops, thanks

**Comment by brcrista** [comment]
PR builds succeeded, but they are still showing pending here.
Linux: https://mseng.visualstudio.com/VSOnline/_build/results?buildId=6700040&view=results
macOS: https://mseng.visualstudio.com/VSOnline/_build/results?buildId=6700040&view=results

**Pull request merged**